### PR TITLE
feat: ProRLv2 - add seq-mask-tis truncated importance sampling type

### PR DESCRIPTION
**Pull request by hijkzzz**
Add a new IS filtering mechanism "seq-mask-tis" that masks entire sequences based on the geometric mean of per-token IS ratios, while keeping non-truncated token-level IS weights for gradient correction. Also adds shared `is_filter_drop_frac` metric for both icepop and seq-mask-tis modes, and documents the new option in prorlv2.md.

## Summary by CodeRabbit

* **New Features**
  * Introduced seq-mask-tis, a sequence-level alternative to ICE-POP for importance sampling that operates at the sequence level rather than per-token.
  * Added is_filter_drop_frac metric for monitoring filtering behavior.

* **Documentation**
  * Comprehensive guide covering seq-mask-tis with rationale, configuration, and comparison with ICE-POP.
  * Updated terminology and expanded examples with new feature references.

### CODE REVIEW

File: docs/guides/prorlv2.md [hunk 1]
```diff
@@ -151,9 +151,38 @@ loss_fn:
...
   - `"icepop"`: set weights outside \([min, max]\) to zero (filter outliers)
+  - `"seq-mask-tis"`: sequence-level geometric-mean mask + non-truncated token-level IS correction (see below)
 
 - **Implementation**: see `ClippedPGLossFn` init-time checks and logic in [`nemo_rl/algorithms/loss_functions.py`](../../nemo_rl/algorithms/loss_functions.py).
 
+### Seq-mask-tis: Sequence-level Geometric-Mean Mask
+
+`seq-mask-tis` is an alternative to ICE-POP that operates at the **sequence level** instead of per-token:
+
+1. For each sequence, compute the **geometric mean** of per-token IS ratios: \(\text{geo\_mean}_i = \exp\!\bigl(\frac{1}{T_i}\sum_t \log \frac{\pi_{\text{prev}}(a_t)}{\pi_{\text{gen}}(a_t)}\bigr)\)
```
Comment (terrykong) [inline docs/guides/prorlv2.md]: is this formula formatted correctly for github and myst-parser?

also how about:

```suggestion
1. For each sequence, compute the **geometric mean** of per-token IS ratios: \(\text{geo\_mean}_i = \exp\!\bigl(\frac{1}{T_i}\sum_t \log \frac{\pi_{\text{train}}(a_t)}{\pi_{\text{gen}}(a_t)}\bigr)\)
```

for clarity

File: nemo_rl/algorithms/loss_functions.py [hunk 2]
```diff
@@ -401,29 +415,63 @@ def __call__(
...
-                    & (
-                        actor_importance_weights_expanded
-                        <= self.truncated_importance_sampling_ratio
-                    ),
-                    actor_importance_weights_expanded,
-                    torch.zeros_like(actor_importance_weights_expanded),
+                    & (seq_geomean_is_ratio <= self.truncated_importance_sampling_ratio)
+                ).float()  # [B]
+                _is_filter_metrics = {
+                    "is_filter_drop_frac": 1.0
```
Comment (terrykong) [inline nemo_rl/algorithms/loss_functions.py]: how about

```suggestion
                    "is_oob_ratio": 1.0
```

so that we can apply it to all importance sampling techniques to know how often it goes out of bounds no matter if filtering or clipping

**Comment by hijkzzz** [comment]
> thanks for contributing @hijkzzz. to help others who come across this PR, is there any experimental results you can share showing how this helps stability in your experiments?
> 
> In the Demystifying blog they show "Token-Level MIS < Sequence-Level MIS" for stability. Any reason why you implemented the token level one instead of the sequence level one first?
> 
> Also, i think it would be good to have unit tests for all these importance sampling techniques for correctness

We found seq-based filtering to be more stable for MoE models.

File: nemo_rl/algorithms/loss_functions.py [hunk 3]
```diff
@@ -401,29 +415,69 @@ def __call__(
...
+                    <= self.truncated_importance_sampling_ratio
+                )
+                _is_filter_metrics = {
+                    "is_oob_ratio": 1.0
+                    - masked_mean(
+                        token_kept_mask.float(),
+                        mask,
+                        global_normalization_factor=global_valid_toks,
+                    ).item(),
+                }
```
Comment (terrykong) [inline nemo_rl/algorithms/loss_functions.py:458]: @hijkzzz thanks for making this change. i think we should also calculate this for `tis` above. this should give users an idea of how often we go out of bounds

**Review by terrykong — APPROVED** [review]
lgtm. @hijkzzz  thanks for making nemo-rl better!
